src/Kobo/SystemDialog.cpp

```diff
@@ -70,8 +73,7 @@ SystemWidget::SwitchKernel()
   KoboModel model = DetectKoboModel();
   if (model == KoboModel::CLARA_HD)
   {
-    ShowMessageBox(_T("This feature is not yet available for Clara HD."), _("Error"),
-                    MB_OK);
+    switch_kernel_button->SetEnabled(0);
```

Comment:
This doesn't make sense. It disables the button *after* the user has successfully pressed it.

Reply:
Stupid me.  I mistakenly didn't test it because I was in git helltrying to fix my pull vs rebase mistake.  A few more minutes...